src/BenchmarkDotNet/Reports/Summary.cs

```diff
@@ -62,13 +63,32 @@ public class Summary
             DisplayPrecisionManager = new DisplayPrecisionManager(this);
             Orderer = GetConfiguredOrdererOrDefaultOne(reports.Select(report => report.BenchmarkCase.Config));
             BenchmarksCases = Orderer.GetSummaryOrder(reports.Select(report => report.BenchmarkCase).ToImmutableArray(), this).ToImmutableArray(); // we sort it first
+            inferredBaselineBenchmarkCase = GetFastestBenchmarkCase(reports);
             Reports = BenchmarksCases.Select(b => ReportMap[b]).ToImmutableArray(); // we use sorted collection to re-create reports list
             BaseliningStrategy = BaseliningStrategy.Create(BenchmarksCases);
             Style = GetConfiguredSummaryStyleOrDefaultOne(BenchmarksCases).WithCultureInfo(cultureInfo);
             Table = GetTable(Style);
             AllRuntimes = BuildAllRuntimes(HostEnvironmentInfo, Reports);
         }
 
+        private static BenchmarkCase GetFastestBenchmarkCase(ImmutableArray<BenchmarkReport> reports)
+        {
+            if (reports.Any() && reports.All(r => r.BenchmarkCase.Config.AutomaticBaselineMode == AutomaticBaselineMode.Fastest))
```

Comment:
```
reports.All(r => r.BenchmarkCase.Config.AutomaticBaselineMode == AutomaticBaselineMode.Fastest))
```

Is it correctly works when 2 benchmark classes have different `AutomaticBaselineMode`?
```C#
[AutomaticBaseline(AutomaticBaselineMode.Fastest)]
public class C1 { }

[AutomaticBaseline(AutomaticBaselineMode.None)]
public class C2 { }
```

To join summaries, use:
1) `--join` (for cmd)
2) `BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).RunAllJoined()`

Reply:
The idea itself works because the benchmarks are already grouped by the type so that Summary contains the cases of either C1 or C2 only. However, this particular example doesn't print the ratio for all the methods. I guess BDN trims columns that don't exist in all results (C1's benchmarks contain ration but C2's ones don't).

Reply:
`ManualConfig.Union(...)` resets the `AutomaticBaselineMode`:

```
config = ManualConfig.Union(config, configFromAttribute.Config) // C1, set the value to "Fastest"
config = ManualConfig.Union(config, configFromAttribute.Config) // C2, set the value to "None" (even if the `[AutomaticBaseline(None)]` is not used)
```

The problem is that the final config has only one value of `AutomaticBaselineMode`, but it should have a different value for each benchmark class to allow:

```C#
[AutomaticBaseline(AutomaticBaselineMode.Fastest)] public class B1 { }
public class B2 { } //implicit AutomaticBaselineMode.None
[AutomaticBaseline(AutomaticBaselineMode.Slowest)] public class B3 { }
```

Reply:
No, it is not like that, the configs are different. I see that this [inferredBaselineBenchmarkCase](https://github.com/dotnet/BenchmarkDotNet/blob/fd9f34fc837d3bebfeca888506e4c6a5fcbec232/src/BenchmarkDotNet/Reports/Summary.cs#L66) is calculated properly. There is probably a need to add a column or something like that, I will try to identify the problem.

Reply:
Yep, you are right. I didn't notice that Summary is built the third time (where you have all 4 methods). I was checking first two instances (first two and second two). Thank you!

Reply:
So, this is what I suggest we do https://github.com/dotnet/BenchmarkDotNet/pull/2171/commits/d20c117b567f0ddc9effd56eeda9c299d97b8ece

1. We skip all defaut (`None`) values to build configs.
2. We take the latest non-default value to determine whether we need to override baseline functionality or we don't.